azkaban-common/src/main/java/azkaban/utils/ThinArchiveUtils.java

```diff
@@ -73,7 +77,8 @@ public static DependencyFile getDependencyFile(final File projectFolder, final D
   public static void writeStartupDependencies(final File f,
       final Set<Dependency> dependencies) throws IOException {
     Map<String, Set<Dependency>> outputFormat = new HashMap<>();
-    outputFormat.put("dependencies", dependencies);
+    Set<Dependency> nonNullDependencies = filterNullFromDeps(dependencies);
+    outputFormat.put("dependencies", nonNullDependencies);
```

Comment:
Can you please replace with outputFormat.put("dependencies", filterNullFromDeps(dependencies));

Reply:
Changed.

Reply:
looks like the changes are not published.